src/guiguts/maintext.py

```diff
@@ -310,6 +329,19 @@ def grid(self, *args: Any, **kwargs: Any) -> None:
         """Override ``grid``, so placing MainText widget actually places surrounding Frame"""
         return self.frame.grid(*args, **kwargs)
 
+    def set_font(self) -> None:
+        """Set the font for the main text widget, based on the current Prefs values."""
+        self.font.config(
+            family=preferences.get(PrefKey.TEXT_FONT_FAMILY),
+            size=preferences.get(PrefKey.TEXT_FONT_SIZE),
+        )
+        # Mac doesn't update window properly, so temporarily select all
+        # then restore selection to force it to update
+        if is_mac():
```

Comment:
This works on Mac:
- Changing font with no selection changes/redraws as you expect. Nothing is selected after.
- With an existing selection or column select, the selection is preserved after the font change.

But on Linux:
- If nothing is selected, the font change works as desired.
- If a selection is active (either columnar or otherwise) then it is lost with the font change.

This code block is guarded by `is_mac()` which I had guessed was the reason, but adding an `is_x11()` test with the below patch did not fix it. Linux still loses the selection on font change.

```diff
--- a/src/guiguts/maintext.py
+++ b/src/guiguts/maintext.py
@@ -14,6 +14,7 @@ import regex as re
 from guiguts.preferences import preferences, PrefKey
 from guiguts.utilities import (
     is_mac,
+    is_x11,
     IndexRowCol,
     IndexRange,
     force_tcl_wholeword,
@@ -337,7 +338,7 @@ class MainText(tk.Text):
         )
         # Mac doesn't update window properly, so temporarily select all
         # then restore selection to force it to update
-        if is_mac():
+        if is_mac() or is_x11():
             ranges = self.selected_ranges()
             self.do_select(IndexRange(self.start(), self.end()))
             self.restore_selection_ranges(ranges)
```

Reply:
Thanks for looking into it @tangledhelix. Earlier on you wrote

> The behavior I describe above does not occur on Ubuntu Linux; there, it behaves as one would expect.

So, I thought you didn't need that code on Linux (or is it a different flavor?)

One option I could try would be to tell it to do the "stash selection, select all, restore selection" after the system becomes idle - no real reason to think that would fix it, but there's no real reason to explain why it's not working.